azurerm/internal/services/synapse/synapse_workspace_resource.go

```diff
@@ -0,0 +1,349 @@
+	}
+	d.Set("name", id.Name)
+	d.Set("resource_group_name", id.ResourceGroup)
+	d.Set("location", location.NormalizeNilable(resp.Location))
+	if err := d.Set("identity", flattenArmWorkspaceManagedIdentity(resp.Identity)); err != nil {
+		return fmt.Errorf("setting identity: %+v", err)
+	}
+	if props := resp.WorkspaceProperties; props != nil {
+		managedVirtualNetworkEnabled := false
+		if props.ManagedVirtualNetwork != nil && strings.EqualFold(*props.ManagedVirtualNetwork, "default") {
+			managedVirtualNetworkEnabled = true
+		}
+		d.Set("managed_virtual_network_enabled", managedVirtualNetworkEnabled)
+		d.Set("storage_data_lake_gen2_filesystem_id", flattenArmWorkspaceDataLakeStorageAccountDetails(props.DefaultDataLakeStorage))
+		d.Set("sql_administrator_login", props.SQLAdministratorLogin)
+		d.Set("managed_resource_group_name", props.ManagedResourceGroupName)
+		d.Set("connectivity_endpoints", utils.FlattenMapStringPtrString(props.ConnectivityEndpoints))
+	}
+	return tags.FlattenAndSet(d, resp.Tags)
+}
+
+func resourceArmSynapseWorkspaceUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Synapse.WorkspaceClient
+	ctx, cancel := timeouts.ForUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.SynapseWorkspaceID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	workspacePatchInfo := synapse.WorkspacePatchInfo{
+		Tags: tags.Expand(d.Get("tags").(map[string]interface{})),
+	}
+	if d.HasChange("sql_administrator_login_password") {
+		workspacePatchInfo.WorkspacePatchProperties = &synapse.WorkspacePatchProperties{
+			SQLAdministratorLoginPassword: utils.String(d.Get("sql_administrator_login_password").(string)),
+		}
+	}
+
+	future, err := client.Update(ctx, id.ResourceGroup, id.Name, workspacePatchInfo)
+	if err != nil {
+		return fmt.Errorf("updating Synapse Workspace %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
+	}
+
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("waiting on updating future for Synapse Workspace %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
+	}
+	return resourceArmSynapseWorkspaceRead(d, meta)
+}
+
+func resourceArmSynapseWorkspaceDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Synapse.WorkspaceClient
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.SynapseWorkspaceID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	if _, err := client.Delete(ctx, id.ResourceGroup, id.Name); err != nil {
+		return fmt.Errorf("deleting Synapse Workspace %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
+	}
+	// sometimes the waitforcompletion rest api will return 404, so changed with polling
+	return waitForSynapseWorkspaceToBeDeleted(ctx, client, id.ResourceGroup, id.Name, d)
```

Comment:
Can we remove the `waitForSynapseWorkspaceToBeDeleted` function and use the SDKs `WaitForCompletionRef`? Just test the response of the for expected values? Maybe something like this:

```
if err = future.WaitForCompletionRef(ctx, client); err != nil {
	if !response.WasNotFound(future.Response()) {
		return fmt.Errorf("waiting for Synapse Workspace %q (Resource Group %q) to be deleted: %+v", name, resourceGroup, err)
	}
}
```